### PR TITLE
OSOE-64: Add samples project to Orchard-Vue.js

### PR DESCRIPTION
**Pull request by sarahelsaig**
OSOE-64

### CODE REVIEW

File: Lombiq.VueJs.Samples/Lombiq.VueJs.Samples.csproj [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  <ItemGroup Condition="'$(NuGetBuild)' != 'true'">
+    <ProjectReference Include="..\Lombiq.VueJs\Lombiq.VueJs.csproj"/>
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(NuGetBuild)' == 'true'">
+    <PackageReference Include="Lombiq.VueJS" Version="1.0.0"/>
+  </ItemGroup>
+
+  <ItemGroup>
+    <EmbeddedResource Include="Assets\Scripts\VueComponents\*.vue"/>
```
Comment (barthamark) [inline Lombiq.VueJs.Samples/Lombiq.VueJs.Samples.csproj]: Is this needed? Don't we have a props file for that?

File: Lombiq.VueJs.Samples/Readme.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# Lombiq Vue.js module for Orchard Core - Samples
+
+
+## About
+
+
+Example Orchard Core module that makes use of Lombiq Vue.js module for Orchard Core.  For general details about and usage instructions see the [root Readme](../Readme.md).
```
Comment (barthamark) [inline Lombiq.VueJs.Samples/Readme.md]:
```suggestion
Example Orchard Core module that makes use of Lombiq Vue.js module for Orchard Core. For general details about and usage instructions see the [root Readme](../Readme.md).
```

File: Lombiq.VueJs.Samples/Readme.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+You can start with either top level sections. Indented sections assume you've already read through their parents.  
+* [Writing Vue.js applications in Orchard Core modules](Controllers/VueAppController.cs)
+* [Writing Vue.js Single File Components in Orchard Core modules](Controllers/VueSfcController.cs)
+  * [Progressive enhancement with SFCs](Controllers/VueSfcController.cs#L31) 
+
+
+## Miscellaneous Notes
+
+You may have noticed that the _package.json_ contains `vue` even thought it's not strictly necessary. It is there to activate the Vue plugin in Rider
```
Comment (barthamark) [inline Lombiq.VueJs.Samples/Readme.md]:
```suggestion
You may have noticed that the _package.json_ contains `vue` even though it's not strictly necessary. It is there to activate the Vue plugin in Rider.
```

File: Lombiq.VueJs.Samples/Lombiq.VueJs.Samples.csproj (hunk 1 above)
Comment (sarahelsaig) [reply Lombiq.VueJs.Samples/Lombiq.VueJs.Samples.csproj]: Not needed. I think I've added the props file later and forgot to update here.

File: Lombiq.VueJs.Samples/Views/VueApp/Index.cshtml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+</div>
+
+@*And finally the initialization. We'd like to pass an options object from here which is tricky since our Vue.js
+application is written using ES6 modules. One way to work around this is to expose an initialization function on the
+window object where we can inject our application options. These options could contain localized texts (assuming that
+you want to localize text only in Orchard Core) and other data like the anti-forgery token if you have forms.*@
+
+<script at="Foot" depends-on="@ResourceNames.DemoApp">
+    window.initDemoApp(@Json.Serialize(new
+                       {
```
Comment (barthamark) [inline Lombiq.VueJs.Samples/Views/VueApp/Index.cshtml]: Is this indentation intentional?

File: Lombiq.VueJs.Samples/wwwroot/vue/demo-repeater.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+'use strict';
```
Comment (barthamark) [inline Lombiq.VueJs.Samples/wwwroot/vue/demo-repeater.js:1]: Why isn't `wwwroot` ignored? Is it related to NuGet packaging?

File: Lombiq.VueJs/Assets/Scripts/helpers/rollup-plugin-vue-sfc-orchard-core.js [hunk 6]
```diff
@@ -76,24 +86,59 @@ module.exports = function vuePlugin() {
...
+                        const result = results[i];
+                        for (let j = 0; j < result.messages.length; j++) {
+                            const message = result.messages[j];
+                            message.line += firstRow - 1;
+                        }
+                    }
+
+                    const formatter = eslint.CLIEngine.getFormatter("stylish");
+
+                    //return JSON.stringify(results, undefined, 2);
```
Comment (barthamark) [inline Lombiq.VueJs/Assets/Scripts/helpers/rollup-plugin-vue-sfc-orchard-core.js]: Leftover code?

**Review by barthamark — CHANGES_REQUESTED** [review]

File: Lombiq.VueJs.Samples/Views/VueApp/Index.cshtml (hunk 4 above)
Comment (sarahelsaig) [reply Lombiq.VueJs.Samples/Views/VueApp/Index.cshtml]: It was auto-formatted like that. I can dedent it.

File: Lombiq.VueJs.Samples/wwwroot/vue/demo-repeater.js (hunk 5 above)
Comment (sarahelsaig) [reply Lombiq.VueJs.Samples/wwwroot/vue/demo-repeater.js:1]: Yes. Because Npm.Targets and Gulp.Extensions are not used in the NuGet.

**Pull request merged**